evm/src/executor/inspector/stack.rs

```diff
@@ -0,0 +1,237 @@
+use super::{Cheatcodes, Debugger, LogCollector, Tracer};
+use bytes::Bytes;
+use ethers::types::{Address, H256};
+use revm::{db::Database, CallInputs, CreateInputs, EVMData, Gas, Inspector, Interpreter, Return};
+
+/// Helper macro to call the same method on multiple inspectors without resorting to dynamic
+/// dispatch
+macro_rules! call_inspectors {
+    ($id:ident, [ $($inspector:expr),+ ], $call:block) => {
+        $({
+            if let Some($id) = $inspector {
+                $call;
+            }
+        })+
+    }
+}
+
+/// An inspector that calls multiple inspectors in sequence.
+///
+/// The order in which inspectors are called match the order in which they were added to the stack.
+///
+/// If a call to an inspector returns a value other than [Return::Continue] (or equivalent) the
+/// remaining inspectors are not called.
+#[derive(Default)]
+pub struct InspectorStack {
+    pub tracer: Option<Tracer>,
+    pub logs: Option<LogCollector>,
+    pub cheatcodes: Option<Cheatcodes>,
+    pub debugger: Option<Debugger>,
+}
+
+impl InspectorStack {
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
+impl<DB> Inspector<DB> for InspectorStack
+where
+    DB: Database,
+{
+    fn initialize_interp(
+        &mut self,
+        interpreter: &mut Interpreter,
+        data: &mut EVMData<'_, DB>,
+        is_static: bool,
+    ) -> Return {
+        call_inspectors!(
```

Comment:
nit: seems like this macro is used a bunch,

I think this always calls the same inspectors, right? followed by a closure

can we instead call all inspectors in a single function instead?

Reply:
Don't think so, the function that is being called on every inspector changes every time. Do you have any suggestions?

Reply:
oh I see the call is invoked for each inspector: `$call;`

and we want to be able to return directly, so we need to expand the code in place, makes sense.